interpreter/src/main/java/org/qbicc/interpreter/impl/VmClassImpl.java

```diff
@@ -514,13 +515,28 @@ public Literal getValueForStaticField(FieldElement field) {
             int val = staticMemory.load8(offset, SinglePlain);
             return lf.literalOf(val != 0);
         } else if (fieldType instanceof SignedIntegerType sit) {
-            return switch (sit.getMinBits()) {
-                case 8 -> lf.literalOf((byte) staticMemory.load8(offset, SinglePlain));
-                case 16 -> lf.literalOf((short) staticMemory.load16(offset, SinglePlain));
-                case 32 -> lf.literalOf(staticMemory.load32(offset, SinglePlain));
-                case 64 -> lf.literalOf(staticMemory.load64(offset, SinglePlain));
-                default -> throw Assert.impossibleSwitchCase(sit.getMinBits());
-            };
+            switch (sit.getMinBits()) {
+                case 8:
+                    return lf.literalOf((byte) staticMemory.load8(offset, SinglePlain));
+                case 16:
+                    return lf.literalOf((short) staticMemory.load16(offset, SinglePlain));
+                case 32:
+                    return lf.literalOf(staticMemory.load32(offset, SinglePlain));
+                case 64: {
+                    try {
+                        return lf.literalOf(staticMemory.load64(offset, SinglePlain));
+                    } catch (InvalidMemoryAccessException e) {
+                        Pointer value = staticMemory.loadPointer(offset, SinglePlain);
+                        if (value == null) {
+                            return lf.zeroInitializerLiteralOfType(sit);
+                        } else {
+                            return lf.valueConvertLiteral(lf.literalOf(value), sit);
+                        }
+                    }
+                }
+                default:
+                    throw Assert.impossibleSwitchCase(sit.getMinBits());
+            }
```

Comment:
```suggestion
            return switch (sit.getMinBits()) {
                case 8 -> lf.literalOf((byte) staticMemory.load8(offset, SinglePlain));
                case 16 -> lf.literalOf((short) staticMemory.load16(offset, SinglePlain));
                case 32 -> lf.literalOf(staticMemory.load32(offset, SinglePlain));
                case 64 -> {
                    try {
                        yield lf.literalOf(staticMemory.load64(offset, SinglePlain));
                    } catch (InvalidMemoryAccessException e) {
                        Pointer value = staticMemory.loadPointer(offset, SinglePlain);
                        if (value == null) {
                            yield lf.zeroInitializerLiteralOfType(sit);
                        } else {
                            yield lf.valueConvertLiteral(lf.literalOf(value), sit);
                        }
                    }
                }
                default -> throw Assert.impossibleSwitchCase(sit.getMinBits());
            }
```
Note: I've never actually used the `yield` syntax but it *should* work...

Reply:
fancy :)  Fixed.